src/test/java/testPackage01/Test_YAMLFileManager.java

```diff
@@ -0,0 +1,254 @@
+package testPackage01;
+
+import com.shaft.driver.SHAFT;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.shaft.validation.Validations.assertThat;
+
+public class Test_YAMLFileManager {
+    static SHAFT.TestData.YAML yaml;
+
+    @BeforeMethod
+    static void setUp() {
+        yaml = new SHAFT.TestData.YAML("src/test/resources/testDataFiles/yaml/yaml_test_data.yaml");
+    }
+
+    @Test
+    static void getObject() {
+        assertThat()
```

Comment:
We should use the built-in assertions to ensure that the end user experience will work as expected, and in case any of the returned datatypes are not handled by the assertion methods of the engine, we should enhance them to ensure consistent behavior.

https://github.com/ShaftHQ/SHAFT_ENGINE/blob/b90e8ff225776bc014f6f57680fc3bf67f646e24/src/test/java/testPackage01/Test_JSONFileManager.java#L10-L14

We should also implement a public method with the same name "getTestData" that has the ability to parse any datatype and return it as a string without the user needing to determine the desired datatype and calling the relevant method.

For that I think the method can be a synonym for getString() in case this one can also handle and parse different data types that are not natively string.

Reply:
Here I'm using the SHAFT assertion in all tests except tests that throw exceptions.

Reply:
Thank you, this is excellent!
Maybe we should add a new validation method for "throws" in a future PR to cover this case